CONTRIBUTING.md

```diff
@@ -8,6 +8,39 @@ don't agree with a convention, submit a PR patching this document and let's disc
 the PR is accepted, *all* code should be updated as soon as possible to reflect the new
 conventions.
 
+Pull Requests
+---
+
+Small, frequent PRs are much preferred to large, infrequent ones. A large PR is difficult
+to review, can block others from making progress, and can quickly get its author into
+"rebase hell". A large PR oftentimes arises when one change requires another, which requires
+another, and then another. When you notice those dependencies, put the fix into a commit of
+its own, then checkout a new branch, and cherrypick it. Open a PR to start the review
+process and then jump back to your original branch to keep making progress. Once the commit
+is merged, you can use git-rebase to purge it from your original branch.
```

Comment:
I would also add that the code should be configured to run in a deployment.  Not just tested locally.  If it's just a local file and tests, consider moving it into a separate directory as a separate module.

Reply:
Sometimes that's quite difficult to do. Like `Bank::hash_internal_state()`, for example, is in the right place, but was waiting on a forking bank before adding it to `Vote`.

Reply:
The hash should have been stuck in a vote and metrics should have reported any inconsistencies.

I think leaving implementations like that in the code creates dead code with tests that need maintaince.  It would be better to target minimal viable testnet functionality. Then we get data on performance and something useful out of it.

Reply:
Added your suggestions, thanks!

Reply:
Thanks 🙏 thanks 😊